lib/pdn/pdn.c

```diff
@@ -600,6 +621,20 @@ static void on_cfun(enum lte_lc_func_mode mode, void *ctx)
 		if (err) {
 			LOG_ERR("Unable to subscribe to +CGEREP=1, err %d", err);
 		}
+
+#if defined(CONFIG_PDN_DEFAULT_AUTH_PAP) || defined(CONFIG_PDN_DEFAULT_AUTH_CHAP)
+		/* +CGAUTH=<cid>[,<auth_prot>[,<userid>[,<password>]]] */
+		BUILD_ASSERT(sizeof(CONFIG_PDN_DEFAULT_USERNAME) > 1, "Username not defined");
+
+		err = pdn_ctx_auth_set(0, CONFIG_PDN_DEFAULT_AUTH,
+				CONFIG_PDN_DEFAULT_USERNAME,
+				CONFIG_PDN_DEFAULT_PASSWORD);
+		if (err) {
+			LOG_ERR("Failed to set auth params for default CID, err %d", err);
+			return;
+		}
+#endif /* CONFIG_PDN_DEFAULT_AUTH_PAP || CONFIG_PDN_DEFAULT_AUTH_CHAP */
```

Comment:
Hang on, in the `CFUN=1` callback, it might be too late to set these parameters, since they configure the default context itself. Should we have a `CFUN=0` callback instead? Otherwise we need to expose the `_init()` function. Might be good to do that regardless since the `pdn` library could be used without the `lte_lc` library.

Reply:
Yeah, that might be too late, indeed... Thanks!